src/xtgeo/grid3d/_gridprop_value_init.py

```diff
@@ -1,101 +1,141 @@
+if TYPE_CHECKING:
+    from xtgeo.grid3d import Grid, GridProperty
 
 
-def gridproperty_non_dummy_values(gridlike, dimensions, values, isdiscrete):
+def gridproperty_non_dummy_values(
+    gridlike: Optional[Union[Grid, GridProperty]],
+    dimensions: Tuple[int, int, int],
+    values: Optional[Union[np.ndarray, float, int]],
+    isdiscrete: bool,
+) -> np.ma.MaskedArray:
     """Gives the initial values array of an gridprop.
 
-    Param:
-      gridlike: Either Grid or GridProperty, giving the mask to replicate.
-      dimensions: The dimensions of the gridprop
-      values: The values parameter given to init.
-      isdiscrete: The discrete parameter given to init.
+    Parameters:
+        gridlike:
+            Either Grid or GridProperty, giving the mask to replicate.
+        dimensions:
+            The (ncol, nrow, nlay) dimensions of the grid property.
+        values:
+            The values parameter given to init.
+        isdiscrete:
+            The discrete parameter given to init.
 
     Returns:
-      The array to be set to GridProp._values.
+        The array to be set to GridProperty._values.
+
     """
     if values is None:
-        values = initial_gridprop_values_zero(dimensions, isdiscrete)
-    elif np.isscalar(values):
-        values = initial_gridprop_values_from_scalar(dimensions, values, isdiscrete)
+        _values = initial_gridprop_values_zero(dimensions, isdiscrete)
+    elif isinstance(values, (int, float)):
+        _values = initial_gridprop_values_from_scalar(dimensions, values, isdiscrete)
     elif isinstance(values, np.ndarray):
-        values = initial_gridprop_values_from_array(dimensions, values, isdiscrete)
+        _values = initial_gridprop_values_from_array(dimensions, values, isdiscrete)
 
-    if gridlike is not None:
+    if gridlike:
         if isinstance(gridlike, xtgeo.grid3d.Grid):
             act = gridlike.get_actnum(asmasked=True)
-            values = np.ma.array(values, mask=np.ma.getmaskarray(act.values))
+            _values = np.ma.array(_values, mask=np.ma.getmaskarray(act.values))
         else:
-            values = np.ma.array(values, mask=np.ma.getmaskarray(gridlike.values))
+            assert isinstance(gridlike, xtgeo.grid3d.GridProperty)
+            _values = np.ma.array(_values, mask=np.ma.getmaskarray(gridlike.values))
 
-    return values
+    return _values
 
 
-def gridproperty_dummy_values(isdiscrete):
+def gridproperty_dummy_values(isdiscrete: bool) -> np.ma.MaskedArray:
     """Given no parameters to init, these dummy values should be set for backwards
-    compatability."""
-    if isdiscrete:
-        values = np.ma.MaskedArray(np.full((4, 3, 5), 99), dtype=np.int32)
-    else:
-        values = np.ma.MaskedArray(np.full((4, 3, 5), 99.0))
+    compatability.
+
+    Parameters:
+        isdiscrete:
+            If the grid property values are discrete.
+
+    Returns:
+        The array to be set to GridProperty._values.
```

Comment:
A `Return type: ...` section is generated by autodoc as inferred from the type annotation. `Returns` should therefore just be a short description of what that return value is.